### PR TITLE
Add createTexture validation tests

### PR DESCRIPTION
**Pull request by beaufortfrancois**
This PR is about adding createTexture validation tests.

FYI @Kangz @austinEng

### CODE REVIEW

File: src/suites/cts/validation/createTexture.spec.ts [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+  const descriptor = t.getDescriptor({ sampleCount, mipLevelCount, arrayLayerCount });
+
+  await t.expectValidationError(() => {
+    t.device.createTexture(descriptor);
+  }, !success);
+}).params([
+  { sampleCount: 1, success: true }, // sampleCount of 1 is allowed
+  { sampleCount: 4, success: true }, // sampleCount of 4 is allowed
+  { sampleCount: 3, success: false }, // sampleCount of 3 is not allowed
```
Comment (kainino0x) [inline src/suites/cts/validation/createTexture.spec.ts]: Should test 8 and 16 as well, at least

File: src/suites/cts/validation/createTexture.spec.ts [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+  { sampleCount: 4, mipLevelCount: 2, success: false }, // it is an error to create a multisampled texture with mipLevelCount > 1.
+  { sampleCount: 4, arrayLayerCount: 2, success: false }, // TODO: Remove when Chrome supports multisampled 2D array textures.
```
Comment (kainino0x) [inline src/suites/cts/validation/createTexture.spec.ts]: Instead we should say success: true, and allow Chrome to be failing the test.

File: src/suites/cts/validation/createTexture.spec.ts [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+  texture.destroy();
+});
+
+g.test('it is invalid to submit a destroyed texture before encode', async t => {
+  const descriptor = t.getDescriptor();
+  const texture = t.device.createTexture(descriptor);
+  const textureView = texture.createView();
+
+  // Destroy texture
+  texture.destroy();
```
Comment (kainino0x) [inline src/suites/cts/validation/createTexture.spec.ts]: nit: I'd parameterize this to make sure the control case works too:

here, use `if (params.destroyBeforeEncode) texture.destroy()`
and similarly for params.destroyAfterEncode, later.

then, cases:
`{ destroyBeforeEncode: false, destroyAfterEncode: false, success: true }`
`{ destroyBeforeEncode: true, destroyAfterEncode: false, success: false }`
`{ destroyBeforeEncode: false, destroyAfterEncode: true, success: false }`

File: src/suites/cts/validation/createTexture.spec.ts [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+
+  await t.expectValidationError(() => {
+    t.device.createTexture(descriptor);
+  }, !success);
+}).params([
+  { format: 'rgba8unorm', success: true }, // rgba8unorm is renderable
+  { format: 'rg11b10float', success: false }, // rg11b10float is not renderable
+  { format: 'rg8snorm', success: false }, // rg8snorm is not renderable
+  { format: 'r8snorm', success: false }, // r8snorm is not renderable
+  { format: 'rgba8snorm', success: false }, // rgba8snorm is not renderable
```
Comment (kainino0x) [inline src/suites/cts/validation/createTexture.spec.ts]: like on the other pr, we should ideally test every texture format.

File: src/suites/cts/validation/createTexture.spec.ts (hunk 1 above)
Comment (beaufortfrancois) [reply src/suites/cts/validation/createTexture.spec.ts]: Done.

File: src/suites/cts/validation/createTexture.spec.ts (hunk 2 above)
Comment (beaufortfrancois) [reply src/suites/cts/validation/createTexture.spec.ts]: Done

File: src/suites/cts/validation/createTexture.spec.ts (hunk 3 above)
Comment (beaufortfrancois) [reply src/suites/cts/validation/createTexture.spec.ts]: Done

File: src/suites/cts/validation/createTexture.spec.ts (hunk 4 above)
Comment (beaufortfrancois) [reply src/suites/cts/validation/createTexture.spec.ts]: Done.

File: src/suites/cts/validation/createTexture.spec.ts [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+}).params([
+  { sampleCount: 0, success: false }, // sampleCount of 0 is not allowed
+  { sampleCount: 1, success: true }, // sampleCount of 1 is allowed
+  { sampleCount: 2, success: false }, // sampleCount of 2 is not allowed
+  { sampleCount: 3, success: false }, // sampleCount of 3 is not allowed
+  { sampleCount: 4, success: true }, // sampleCount of 4 is allowed
+  { sampleCount: 8, success: false }, // sampleCount of 8 is not allowed
+  { sampleCount: 16, success: false }, // sampleCount of 16 is not allowed
+  { sampleCount: 4, mipLevelCount: 2, success: false }, // it is an error to create a multisampled texture with mipLevelCount > 1
+  { sampleCount: 4, arrayLayerCount: 2, success: true }, // multisampled 2D array texture is not supported
```
Comment (kainino0x) [inline src/suites/cts/validation/createTexture.spec.ts]:
```suggestion
  { sampleCount: 4, arrayLayerCount: 2, success: true }, // multisampled 2D array texture is supported
```

**Pull request merged**